Makefile

```
@@ -889,6 +889,8 @@ update-js: node-check | node_modules
 	npx updates -u -f package.json
 	rm -rf node_modules package-lock.json
 	npm install --package-lock
+	npx nolyfill install
+	npm update
```

```suggestion
  npx nolyfill install
	npm install --package-lock
```

- We need  to run it before `npm install` so that the changes take effect
- We never want to run `npm update`, we have `npx updates` to do the same job

---

npm update is required because of an upstream bug where npm install doesn't respect the new field in package.json (the cli output shows the link to the bug)

---

It links to https://github.com/npm/cli/issues/5850 which says the bug manifests only for non-first installs and first install is when `package-lock.json` is absent which is the case here as we delete it a line before. So O don't think my version is affected by that bug.

``npm update` at this point is pointless, except that is has the danger of upgrading "locked" dependencies.

---

@silverwind awesome. thanks for looking into that :) I will update with your suggestion above.

---

To slightly correct myself: "First install" scenario is when both `package-lock.json` and `node_modules` are absent which we guarantee above. Reason we do it that way is also to avoid https://github.com/npm/cli/issues/4828. It seems npm has a number of unresolved bugs when not "first install" 😆.

---

Actually I will give my suggestion a try because I'm not sure it's ideal to run `npx nolyfill` with no node_modules present, likely it will work but npx would install it to some temporary location.

---

@silverwind were you able to try that out? If not, I can make the above change and then try it myself.

---

Suggestion did not work because `npx` would prompt to install temporary package. So I ended up running `npm install` twice now which is ugly but works.